tests/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Razor.Tests/DefaultTextMatcherTests.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using DiffPlex;
+using DiffPlex.Chunkers;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Steps.Razor.Tests
+{
+    public class DefaultTextMatcherTests
+    {
+        [Fact]
+        public void CtorNegativeTests()
+        {
+            Assert.Throws<ArgumentNullException>("differ", () => new DefaultTextMatcher(null!, new CharacterChunker()));
+            Assert.Throws<ArgumentNullException>("chunker", () => new DefaultTextMatcher(new Differ(), null!));
+        }
+
+        [Fact]
+        public void MatchOrderedSubTextsNegativeTests()
+        {
+            var matcher = new DefaultTextMatcher(new Differ(), new CharacterChunker());
+            var originalTexts = new MappedSubText[]
+            {
+                GetSubText("Test", "test.txt", 1)
+            };
+            var updatedTexts = new string[] { "Test1", "Test2" };
+
+            Assert.Throws<ArgumentNullException>("originalTexts", () => matcher.MatchOrderedSubTexts(null!, updatedTexts));
+            Assert.Throws<ArgumentNullException>("newTexts", () => matcher.MatchOrderedSubTexts(originalTexts, null!));
+            Assert.Throws<ArgumentException>(() => matcher.MatchOrderedSubTexts(originalTexts, updatedTexts));
+        }
+
+        [Theory]
+        [MemberData(nameof(MatchOrderedSubTextsData))]
+        public void MatchOrderedSubTextsPositiveTests(IEnumerable<MappedSubText> originalTexts, IEnumerable<string> newTexts, IEnumerable<TextReplacement> expectedReplacements)
+        {
+            // Arrange
+            var matcher = new DefaultTextMatcher(new Differ(), new CharacterChunker());
+
+            // Act
+            var replacements = matcher.MatchOrderedSubTexts(originalTexts, newTexts);
+
+            // Assert
+            Assert.Collection(replacements, expectedReplacements.Select<TextReplacement, Action<TextReplacement>>(expected => actual => Assert.Equal(expected, actual)).ToArray());
+        }
+
+        public static IEnumerable<object[]> MatchOrderedSubTextsData =>
+            new List<object[]>
```

Comment:
nit: I've often used a `yield return` syntax here. Seems like that would be slightly more readable

Reply:
Good idea. I'll try that.